server/chat-plugins/net-filters.ts

```diff
@@ -0,0 +1,217 @@
+/**
+ * Neural net chatfilters.
+ * These are in a separate file so that they don't crash the other filters.
+ * (issues with globals, etc)
+ * by Mia.
+ * @author mia-pi-git
+ */
+
+import {QueryProcessManager, QueryProcessWrapper} from '../../lib/process-manager';
+import {FS} from '../../lib/fs';
+import {Utils} from '../../lib/utils';
+import {Config} from '../config-loader';
+
+const PATH = "config/chat-plugins/net.json";
+const MAX_PROCESSES = Config.netfilterprocesses || 1;
+const PM_TIMEOUT = 2 * 60 * 60 * 1000; // training can be _really_ slow
+const WHITELIST = ["mia"];
+
+interface NetQuery {
+	data: string | TrainingLine[];
+	type: "run" | "train" | "save" | "load";
+	options?: AnyObject;
+}
+
+interface TrainingLine {
+	input: string;
+	// "|flag" (bad) "|ok" (good)
+	output: string;
+}
+
+export class NeuralNetChecker {
+	model: import('brain.js').recurrent.LSTM | null;
+	constructor(path?: string) {
+		try {
+			this.model = new (require('brain.js').recurrent.LSTM)();
+		} catch (e) {
+			this.model = null;
+		}
+		if (path) this.load(path);
+	}
+	async train(data: TrainingLine[], iterations?: number) {
+		// 100 has good perf but is still effective
+		if (!iterations) iterations = 100;
+		const now = Date.now();
+		await FS(PATH).copyFile(PATH + '.backup');
+		this.model?.train(data, {iterations});
```

Comment:
something something design principle silent failure

Reply:
Modified this in the checkAllowed() function so that it should always loudly fail in the command if it exists.